plonky2/src/recursion/cyclic_recursion.rs

```diff
@@ -183,10 +183,10 @@ pub fn set_cyclic_recursion_data_target<
     const D: usize,
 >(
     pw: &mut PartialWitness<F>,
-    cyclic_recursion_data_target: &CyclicRecursionTarget<D>,
+    cyclic_recursion_data_target: &CyclicRecursionTarget<F, C, D>,
     cyclic_recursion_data: &CyclicRecursionData<F, C, D>,
     // Public inputs to set in the base case to seed some initial data.
-    public_inputs: &[F],
+    public_inputs: &[(usize, F)],
```

Comment:
nit: Maybe comment on the format of `public_inputs` here (e.g like the comment for `dummy_proof()`) since this is a public function.